src/mlx/warnings.py

```diff
@@ -13,45 +14,99 @@
 
 
 class WarningsPlugin:
-    def __init__(self):
-        self.sphinx_counter = 0
-        self.doxygen_counter = 0
-        self.junit_counter = 0
 
-    def check_sphinx_warnings(self, line):
+    def __init__(self, sphinx = False, doxygen = False, junit = False):
```

Comment:
This does not seem very scaleable to me.

Could it be an idea to not do it in the constructor itself, but add a function called ```append_checker(str)```?

``` py

def append_checker(self, checker_name):
    self.checkerList.append(WarningsChecker(checker_name, PATTERN_LUT[checker_name]))
```